src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -65,6 +65,22 @@ public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
                     {
                         ConfigureTempApiHashKey(builder, settings.Authentication);
                     }
+
+                    // User-specified configuration file path is considered highest precedence, but does NOT override other configuration sources
+                    string userFilePath = settings.UserProvidedConfigFilePath;
+
+                    if (File.Exists(userFilePath) && Path.GetExtension(userFilePath).ToLower().Equals(".json"))
+                    {
+                        try
+                        {
+                            File.OpenRead(userFilePath).Dispose(); // If this succeeds, we have read-permissions
+                            builder.AddJsonFile(userFilePath, optional: true, reloadOnChange: true);
+                        }
+                        catch (Exception)
+                        {
+                            // Don't currently log to tell the user that their configuration file was not loaded
```

Comment:
Take a look at `AddressListenResults` to see how to transfer error state from here to where a logger is available.

Reply:
To be able to mimic that, would we want/need some sort of wrapper around `IHostBuilder` that also includes an error list? And then the caller (e.g. `CollectCommandHandler`) would be responsible for iterating over the error list and logging it using the service from the host?

Reply:
I was under the mistaken impression that services are available in this context, but this is just the configuration block. Let me try to play around with the host builder context to see if this is easy to do.

Reply:
One option here would be to use the `ConfigureAppConfiguration` overload that takes a `HostBuilderContext` object. You can then set `HostBuilderContext.Properties` to include errors.

Then during `Startup.Configure` you can add `HostBuilderContext` as a parameter, and retrieve the errors from `Properties`.

Reply:
I have a version of this working, but wanted to double check what you meant in terms of the implementation - is there an actual way to set errors for `Properties`, or did you just mean that I can just add the error as a property (or a list of errors for a key that holds all the errors) and print out the `Properties` from `Startup.Configure`?

Reply:
The latter - Properties here is just key/value pairs that are used to communicate information between various parts of the host building process. Normally we would use a singleton object (like AddressListenResults) but I'm not sure it's possible to add a singleton this early in the host builder.